pkg/target/target_builder.go

```diff
@@ -164,49 +164,58 @@ func (b *targetBuilderImpl) SetShoot(ctx context.Context, name string) TargetBui
 			return ErrNoGardenTargeted
 		}
 
-		gardenClient, err := b.getGardenClient(t.GardenName())
-		if err != nil {
-			return err
-		}
+		return b.completeTargetForShoot(ctx, t, name)
+	})
 
-		shoot, err := gardenClient.FindShoot(ctx, t.WithShootName(name).AsListOption())
-		if err != nil {
-			return fmt.Errorf("failed to fetch shoot: %w", err)
-		}
+	return b
+}
 
-		if t.ProjectName() == "" {
-			// we need to resolve the project name as it is not already set
-			// This is important to ensure that the target stays unambiguous and the shoot can be found faster in subsequent operations
-			project, err := gardenClient.GetProjectByNamespace(ctx, shoot.Namespace)
-			if err != nil {
-				return fmt.Errorf("failed to fetch parent project for shoot: %w", err)
-			}
+func (b *targetBuilderImpl) SetControlPlane(ctx context.Context) TargetBuilder {
+	b.actions = append(b.actions, func(t *targetImpl) error {
+		if t.GardenName() == "" {
+			return ErrNoGardenTargeted
+		}
 
-			t.Project = project.Name
+		err := b.completeTargetForShoot(ctx, t, t.ShootName())
+		if err != nil {
+			return err
 		}
 
-		t.Seed = ""
-		t.Shoot = shoot.Name
-		t.ControlPlaneFlag = false
+		t.ControlPlaneFlag = true
 
 		return nil
 	})
 
 	return b
 }
 
-func (b *targetBuilderImpl) SetControlPlane(ctx context.Context) TargetBuilder {
-	b.actions = append(b.actions, func(t *targetImpl) error {
-		if t.Shoot == "" {
-			return ErrNoShootTargeted
+func (b *targetBuilderImpl) completeTargetForShoot(ctx context.Context, t *targetImpl, name string) error {
+	gardenClient, err := b.getGardenClient(t.GardenName())
+	if err != nil {
+		return err
+	}
+
+	shoot, err := gardenClient.FindShoot(ctx, t.WithShootName(name).AsListOption())
+	if err != nil {
+		return fmt.Errorf("failed to fetch shoot: %w", err)
+	}
+
+	if t.ProjectName() == "" {
```

Comment:
```suggestion
	if t.Project == "" {
```

Reply:
Maybe also for GardenName because we need to use targetImpl here.

Reply:
Yes, I was wondering the same - so let's always use the property I will change it